### ISSUE TITLE
Cleanup Assert.True(false) in Xml tests

### ISSUE THREAD

**Issue by sepidehkh** [dotnet/corefx#13647]
In case of failure, Xml tests are doing Assert.True(false), which doesn't help in seeing what the problem and expected behaviour was. Use Assert.Throws<EXPECTED_EXCEPTION> instead wherever applies.

**Comment by Clockwork-Muse** [comment]
Ah!  Thanks for working on this.  I have some [stuff I did in this space](https://github.com/Clockwork-Muse/corefx/tree/System.Xml.XmlSchemaSet), but got distracted by something else.

Note that beyond just the exceptions, the tests have some problems.  The validation event handler used in many tests is mentioned multiple times in the same test, making it impossible to determine _where_ a specific test has failed.  They also use static variables, which may cause problems in some parallel scenarios (although xunit doesn't run the whole test class in parallel - currently - it's probably better to keep all such data contained anyways).  There are swapped variables (expected/actual), additional custom error handlers, and possibly more.  Dunno how much you were planning on doing for this...

**Comment by sepidehMS** [comment]
@Clockwork-Muse Thank you for pointing these out and your work on this. Please feel free to submit a PR for fixing some of these issues.

**Comment by krwq** [comment]
Closing this as won't fix. Improvement is welcome but it probably makes sense to fix it when spotted than spending time on this.
